lib/document.js

```diff
@@ -2938,15 +2940,24 @@ function _pushNestedArrayPaths(val, paths, path) {
  * ignore
  */
 
-Document.prototype._execDocumentPreHooks = async function _execDocumentPreHooks(opName, ...args) {
-  return this.$__middleware.execPre(opName, this, [...args]);
+Document.prototype._execDocumentPreHooks = async function _execDocumentPreHooks(opName, options, ...args) {
+  if (shouldSkipMiddleware(options, 'pre')) {
+    return [options, ...args];
+  }
+  return this.$__middleware.execPre(opName, this, [options, ...args]);
```

Comment:
similarly potentially backwards breaking here - adding `options` to all document pre hooks.